README.md
```
############      #######       ##########       #####     ######
#############     #######      ###########       #####    ######
####     #####    #######     #####              #####   ######
####     #####    #######     ######             #####  ######
####    ####      #######      ###########       ############
###########       #######       ###########      ############
#####  ######     #######             #####      ###### #######
#####   ######    #######            ######      ######  #######
#####    ######   #######     ###########        ######   #######
#####    ######   #######     ##########         ######    #######
```

# Risk: The Game of World Domination
##### (Racket Edition)

##### Version 1.8a

(**Note to Developers**: Racket v6.3 is **required** to run this program. The latest version can be found [here](https://download.racket-lang.org/))

Project Name - Risk: *Racket Edition*

Included libraries, modules, and definitions files:
- The ```src``` library contains ALL source code used in the game.
  - The ```imgs``` library contains all images used in the game.
  - The ```bkup``` library contains all back-up files for program files.
  - The ```fonts``` library contains fonts used in by Racket to draw text with that font.
  - The ```sounds``` library contains mp3 source files that can play during the game using rsound, an extension to Racket that is not yet implemented in Risk.
  - The ```testfiles``` library contains programs that test functions not easily tested by check-expects.
  - The ```Map-Positions``` library contains files that store the information of hitboxes used in territories in-game.
    - ```map-positions.txt``` contains a list of hitboxes for all territories.

  (**Note**: The ```compiled``` library is often created by the DrRacket Racket IDE when creating errortrace files, but is not tracked by the repository using ```.gitignore```.)

  - The ```dice-functs.rkt``` module contains functions to help simulate dice rolls during gameplay for things such as turn selection and attacks/defenses.
  - The ```graph.rkt``` module contains functions that are used for animation purposes such as point comparisons.
  - The ```ellipse.rkt``` module provides important ellipse functions used for hitboxes and math throughout programs.
  - The ```hit-box-determiner.rkt``` program allows hitboxes to be made for all territories in the game using ellipses, and outputs results to a file inside of the ```Map-Positions``` lib.
  - ```test-animations.rkt``` contains tests for any animations thought to require tests outside of their host files.
  - ```matdes.rkt``` module contains material design functions used to construct the UI.
  - ```sys-lists.rkt``` contains constants used to power the system model.
  - ```riskFunctions.rkt``` is the main thread that calls all of the modules above to create Risk, and is used to create the executable for any build.

## Preferred Syntax

To require particular files, such as the modules listed above, using the Racket language follows the syntax:
```racket
(require [String path])
```

To require a library follows the syntax:
```racket
(require [identifier lib-name])
```

To provide functions for use in other programs upon request in Racket uses the syntax:
```racket
(provide function-name-1 ... function-name-x)
```

To provide a struct in Racket uses the syntax:
```racket
(provide (struct-out struct-id))
```

To provide all identifiers in a given file uses the syntax:
```racket
(provide (all-defined-out))
```

Images in the "imgs" library will be utilized with the bitmap function provided by Racket, with the syntax:
```racket
(bitmap [String path])
```
These images will be defined as global variables in the source code that they are used in.

## Contributing
The project is located [here](https://github.com/Viviose/Risk.git).
Persons interested in being an official member of this team can contact the developers at any time for more information. This will allow users read and write access to the repo at any time.
Anyone that wishes to contribute anything to this project can fork the project and create a pull request that will be reviewed before implementation.
Game files can be downloaded into any folder with any path, but to be updated remotely, they must have Git initialized in said folder.
The best way to do this is to clone the repo using Git with the following command:

```
git clone https://github.com/Viviose/Risk.git
```

The home page for the project can be found [here](https://github.com/Viviose/Risk).

## Installation
To build the game executable on Windows, run the build.bat batch file located in the root game folder.
To build the game executable on Mac/Linux, run the build.sh shell script located in the root game folder.

## Authors
Joshua Sanchez & Patrick "Patty G" Gallagher

## Project Management Features
Bugs should be reported [here](https://github.com/Viviose/Risk/issues).

The wiki can be found [here](https://github.com/Viviose/Risk/wiki).

## Contact
Josh Sanchez (Project Manager and Head Programmer)
- E-mail: joshuaesanch@gmail.com
- Phone Number: +1(773)499-5535

Patty G (Head Programmer)
- E-mail: pghahaha@cpgallagher.com
- Phone Number: +1(312)493-5743


changelog.txt

BUILD IS NOW IN ALPHA
Current Version: 1.8a
Version 1.8a (1/11/15)
- Added in a mini-mode for smaller screens (Under heavy development)
- Made a fullscreen mode
- Updated project to require Racket v6.3

Version 1.7a (12/28/15)
- Made the window dynamic to the user's screen size
- Fixed hitboxes to be dynamic as well
- Made new constant variables to help with positioning

Version 1.6.12a (12/27/15)
- Scrapping existing dice-functs functions for redo
- Implementing new structs for dice and new functions for compatibility and usage

Version 1.6.11a (12/2/15)
- Added conditionals to attack-phase
- Added win clause to attack-phase
- Added conditional index for attack-phase

Version 1.6.10a "The Developer's Update" (12/1/15)
- Updated README to match latest project info, including links, more modules, and version update
- Updated DevVLog to much better reflect progress each commit period in order provide a more useful development history
- Updated back-up files
- Added new-line to .gitignore file

Version 1.6.9a (11/30/15)
- Fixed issue in attack-phase clause (https://github.com/Viviose/Risk/issues/3)

Version 1.6.8a (11/29/15)
- .gitignored errortrace files
- Defined SLIDER-OFFSET constant for ease in redefining slider hitboxes and implemented in appropriate locations legal
- Updated calc-armies function
- Added in conditions for territory bordering in attack-phase clauses
- Added an error message for render for use in debugging that alerts users when render breaks due to incorrect screens (https://github.com/Viviose/Risk/issues/2)
- Fixed bugs occurring as a result of screen issues (https://github.com/Viviose/Risk/issues/5)
- Fixed inconsistencies in spelling of keywords in card lists
- Added tick handler for potential use in animations for dice and attacks, results in heavy lag as of now
- jash is puu

Version 1.6.7a (11/28/15)
- Added SLIDER-WARN constant to render and matdes.rkt to remind players to select troops before launching attack
- Moved INITIAL-CARD-LIST, INITIAL-TERRITORY-LIST, and DEBUG-CARD-LIST to sys-lists.rkt, which stores these constants
- Moved card and territory structs to sys-lists.rkt and provided/required them as necessary for compatibility
- Moved textc roboto function into matdes.rkt
- Updated DEBUG-TERRITORY-LIST for debugging purposes
- Added "slider_warning" screen to implement SLIDER-WARN
- Updated conditionals in attack-phase
- Finished first functioning build of attack-phase (definitely not free of bugs)

Version 1.6.6a (11/27/15)
- Added return values for some attack-phase clauses
- Added territory-attacking and territory-attacked parameters to system struct to streamline comparisons in attack-phase

***MOVED PROJECT TO GITHUB 11/24/15***

Version 1.6.5a (11/23/15)
- Removed deprecated parameters from structs
- Reworked initial troop allocation to account for deprecated parameters
- Added victory checking conditions to attack phase
- Finalized debug button in key-handler **ONLY FOR USE IN 5 PLAYERS MODE**

Version 1.6.4a (11/21/15)
- Added a catch for which-card function that allows no cards to be selected without errors
- Fixed conditional bug that prevented card selection
- Added catch for change-card-selected function that ran into issues when the card-list was empty
- Created a function that returns the cards owned by a player as a list
- Began debug button clause for key-handler

Version 1.6.3a (11/20/15)
- Added armies-attacking parameter to system struct in order to count how many armies are being used in an attack, used in creating and implementing slider
- Fixed more bugs in recruit-phase, creating a stable usage
- Created a function that grabs territories from the territory list given the model
- Added in conditions for attack-phase
- Defined initial value for armies-attacking in big-bang

Version 1.6.2a (11/18/15)
- Fixed function return bugs in recruit-phase
- Improved slider struct to make it function more dynamically and stable

Version 1.6.1a (11/17/15)
- Moved between? discrminator to graph.rkt
- Began work on attack-phase
- Added key handler that closes the game on pressing the escape button

Version 1.6a (11/15/15)
- Finished first functioning usage of recruit-phase

Version 1.5.3a (11/1/15)
- Finished card redemption functions, and implemented them in the appropriate locations
- Fixed bugs in recruit-phase and added final clauses

Version 1.5.2a (10/26/15)
- Created more card redemption functions that set cards into an active state when clicked on and turn in cards
- Fixed bugs relating to transition between init-recruit and recruit phase

Version 1.5.1a (10/19/15)
- Added troop recruitment to the end of the initial-recruit phase so that P1 starts recruit with appropriate troops
- Finished first clause of recruit-phase

Version 1.5a (10/18/15)
- Created functions to check card-list for types and quantities
- Created functions that check card-list for conditions concerning type and quantity

Version 1.4.4a (8/17/15)
- Aligned cards to the left
- Gathered cards hitboxes
- Created a function similar to territory-update for updating cards' owners and states.

Version 1.4.3a (7/17/15)
- Added in state parameter to card struct for us in card functions and related drawing functions, letting the system know a particular card has been chosen.
- Updated INITIAL-CARD-LIST to include State parameters for each card, defaulted to "inactive"

Version 1.4.2a (7/11/15)
- Added in id parameter to card struct for use in randomly attributing cards to players after finishing recruit phase.
- Updated INITIAL-CARD-LIST to include ID parameters for each card, ranging from 0-42.
- Created function that allows for the program to create a list of cards a player owns.

Version 1.4.1a (7/8/15)
- Added in base territory army recruitment calculator
- Moved cardlist variable from player struct to system struct
- Added INITIAL-CARD-LIST to store all card info, including Territory, Owner, and Unit Type
- Started Card redemption functions

Version 1.4a (7/4/15)

src/test-animations.rkt
#reader(lib"read.ss""wxme")WXME0108 ## 
#|
   This file uses the GRacket editor format.
   Open this file in DrRacket version 6.1 or later to read it.

   Most likely, it was created by saving a program in DrRacket,
   and it probably contains a program with non-text elements
   (such as images or comment boxes).

            http://racket-lang.org/
|#
 31 7 #"wxtext\0"
3 1 6 #"wxtab\0"
1 1 8 #"wximage\0"
2 0 8 #"wxmedia\0"
4 1 34 #"(lib \"syntax-browser.ss\" \"mrlib\")\0"
1 0 16 #"drscheme:number\0"
3 0 44 #"(lib \"number-snip.ss\" \"drscheme\" \"private\")\0"
1 0 36 #"(lib \"comment-snip.ss\" \"framework\")\0"
1 0 93
(
 #"((lib \"collapsed-snipclass.ss\" \"framework\") (lib \"collapsed-sni"
 #"pclass-wxme.ss\" \"framework\"))\0"
) 0 0 43 #"(lib \"collapsed-snipclass.ss\" \"framework\")\0"
0 0 19 #"drscheme:sexp-snip\0"
0 0 36 #"(lib \"cache-image-snip.ss\" \"mrlib\")\0"
1 0 68
(
 #"((lib \"image-core.ss\" \"mrlib\") (lib \"image-core-wxme.rkt\" \"mr"
 #"lib\"))\0"
) 1 0 29 #"drscheme:bindings-snipclass%\0"
1 0 88
(
 #"((lib \"pict-snip.rkt\" \"drracket\" \"private\") (lib \"pict-snip.r"
 #"kt\" \"drracket\" \"private\"))\0"
) 0 0 34 #"(lib \"bullet-snip.rkt\" \"browser\")\0"
0 0 25 #"(lib \"matrix.ss\" \"htdp\")\0"
1 0 22 #"drscheme:lambda-snip%\0"
1 0 29 #"drclickable-string-snipclass\0"
0 0 26 #"drracket:spacer-snipclass\0"
0 0 57
#"(lib \"hrule-snip.rkt\" \"macro-debugger\" \"syntax-browser\")\0"
1 0 26 #"drscheme:pict-value-snip%\0"
0 0 45 #"(lib \"image-snipr.ss\" \"slideshow\" \"private\")\0"
1 0 38 #"(lib \"pict-snipclass.ss\" \"slideshow\")\0"
2 0 55 #"(lib \"vertical-separator-snip.ss\" \"stepper\" \"private\")\0"
1 0 18 #"drscheme:xml-snip\0"
1 0 31 #"(lib \"xml-snipclass.ss\" \"xml\")\0"
1 0 21 #"drscheme:scheme-snip\0"
2 0 34 #"(lib \"scheme-snipclass.ss\" \"xml\")\0"
1 0 10 #"text-box%\0"
1 0 32 #"(lib \"text-snipclass.ss\" \"xml\")\0"
1 0 1 6 #"wxloc\0"
          0 0 55 0 1 #"\0"
0 75 1 #"\0"
0 12 90 -1 90 -1 3 -1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 255 255 255 1 -1 0 9
#"Standard\0"
0 75 10 #"Monospace\0"
0 12 90 -1 90 -1 3 -1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 255 255 255 1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 1 1 1 1 1 1 0 0 0 0 0 0 -1 -1 2 24
#"framework:default-color\0"
0 -1 1 #"\0"
1 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 255 255 255 -1 -1 2
1 #"\0"
0 -1 1 #"\0"
1 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 0 0 0 1 1 1 150 0 150 0 0 0 -1 -1 2 15
#"text:ports out\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 150 0 150 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1.0 0 -1 -1 93 -1 -1 -1 0 0 0 0 0 0 0 0 0 1.0 1.0 1.0 255 0 0 0 0 0 -1
-1 2 15 #"text:ports err\0"
0 -1 1 #"\0"
1 0 -1 92 93 -1 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 255 0 0 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 0 0 0 1 1 1 0 0 175 0 0 0 -1 -1 2 17
#"text:ports value\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 175 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1.0 0 92 -1 -1 -1 -1 -1 0 0 0 0 0 0 0 0 0 1.0 1.0 1.0 34 139 34 0 0 0 -1
-1 2 27 #"Matching Parenthesis Style\0"
0 -1 1 #"\0"
1.0 0 92 -1 -1 -1 -1 -1 0 0 0 0 0 0 0 0 0 1.0 1.0 1.0 34 139 34 0 0 0 -1
-1 2 1 #"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 38 38 128 0 0 0 -1 -1 2 37
#"framework:syntax-color:scheme:symbol\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 38 38 128 0 0 0 -1 -1 2 38
#"framework:syntax-color:scheme:keyword\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 38 38 128 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 194 116 31 0 0 0 -1 -1 2
38 #"framework:syntax-color:scheme:comment\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 194 116 31 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 41 128 38 0 0 0 -1 -1 2 37
#"framework:syntax-color:scheme:string\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 41 128 38 0 0 0 -1 -1 2 35
#"framework:syntax-color:scheme:text\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 41 128 38 0 0 0 -1 -1 2 39
#"framework:syntax-color:scheme:constant\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 41 128 38 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 132 60 36 0 0 0 -1 -1 2 49
#"framework:syntax-color:scheme:hash-colon-keyword\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 132 60 36 0 0 0 -1 -1 2 42
#"framework:syntax-color:scheme:parenthesis\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 132 60 36 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 255 0 0 0 0 0 -1 -1 2 36
#"framework:syntax-color:scheme:error\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 255 0 0 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 -1 -1 2 36
#"framework:syntax-color:scheme:other\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 -1 -1 2 16
#"Misspelled Text\0"
0 -1 1 #"\0"
1 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 81 112 203 0 0 0 -1 -1 2
38 #"drracket:check-syntax:lexically-bound\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 81 112 203 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 178 34 34 0 0 0 -1 -1 2 28
#"drracket:check-syntax:set!d\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 178 34 34 0 0 0 -1 -1 2 37
#"drracket:check-syntax:unused-require\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 255 0 0 0 0 0 -1 -1 2 36
#"drracket:check-syntax:free-variable\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 255 0 0 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 68 0 203 0 0 0 -1 -1 2 31
#"drracket:check-syntax:imported\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 68 0 203 0 0 0 -1 -1 2 47
#"drracket:check-syntax:my-obligation-style-pref\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 178 34 34 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 116 0 0 0 0 -1 -1 2 50
#"drracket:check-syntax:their-obligation-style-pref\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 116 0 0 0 0 -1 -1 2 48
#"drracket:check-syntax:unk-obligation-style-pref\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 139 142 28 0 0 0 -1 -1 2
49 #"drracket:check-syntax:both-obligation-style-pref\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 139 142 28 0 0 0 -1 -1 2
26 #"plt:htdp:test-coverage-on\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 -1 -1 2 1
#"\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 1 0 0 0 0 0 0 255 165 0 0 0 0 -1 -1 2 27
#"plt:htdp:test-coverage-off\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 1 0 0 0 0 0 0 255 165 0 0 0 0 -1 -1 4 1
#"\0"
0 70 1 #"\0"
1.0 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 1.0 1.0 1.0 1.0 1.0 1.0 0 0 0 0 0 0
-1 -1 4 4 #"XML\0"
0 70 1 #"\0"
1.0 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 1.0 1.0 1.0 1.0 1.0 1.0 0 0 0 0 0 0
-1 -1 2 37 #"plt:module-language:test-coverage-on\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 -1 -1 2 38
#"plt:module-language:test-coverage-off\0"
0 -1 1 #"\0"
1 0 -1 92 -1 93 -1 -1 0 1 0 0 0 1 0 0 0 0 0 0 255 165 0 0 0 0 -1 -1 4 1
#"\0"
0 71 1 #"\0"
1.0 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 1.0 1.0 1.0 1.0 1.0 1.0 0 0 0 0 0 0
-1 -1 4 1 #"\0"
0 -1 1 #"\0"
1.0 0 -1 -1 -1 -1 -1 -1 1 0 0 0 0 0 0 0 0 1.0 1.0 1.0 0 0 255 0 0 0 -1
-1 4 1 #"\0"
0 71 1 #"\0"
1.0 0 -1 -1 -1 -1 -1 -1 1 0 0 0 0 0 0 0 0 1.0 1.0 1.0 0 0 255 0 0 0 -1
-1 4 1 #"\0"
0 71 1 #"\0"
1.0 0 -1 -1 -1 -1 -1 -1 0 0 0 0 0 0 0 0 0 1.0 1.0 1.0 0 100 0 0 0 0 -1
-1           0 124 0 28 3 12 #"#lang racket"
0 0 24 29 1 #"\n"
0 0 24 3 1 #"("
0 0 15 3 7 #"require"
0 0 24 3 1 #" "
0 0 14 3 18 #"picturing-programs"
0 0 24 3 1 #")"
0 0 24 29 1 #"\n"
0 0 24 29 1 #"\n"
0 0 24 3 1 #"("
0 0 15 3 7 #"require"
0 0 24 3 1 #" "
0 0 19 3 11 #"\"graph.rkt\""
0 0 24 3 1 #")"
0 0 24 29 1 #"\n"
0 0 24 29 1 #"\n"
0 0 24 3 1 #"("
0 0 15 3 6 #"define"
0 0 24 3 2 #" ("
0 0 14 3 6 #"render"
0 0 24 3 1 #" "
0 0 14 3 5 #"model"
0 0 24 3 1 #")"
0 0 24 29 1 #"\n"
0 0 24 3 3 #"  ("
0 0 15 3 4 #"cond"
0 0 24 3 3 #" [("
0 0 14 3 6 #"equal?"
0 0 24 3 1 #" "
0 0 14 3 5 #"model"
0 0 24 3 1 #" "
0 0 21 3 2 #"#t"
0 0 24 3 1 #")"
0 0 24 29 1 #"\n"
0 0 24 3 10 #"         ("
0 0 14 3 9 #"rectangle"
0 0 24 3 1 #" "
0 0 21 3 3 #"700"
0 0 24 3 1 #" "
0 0 21 3 3 #"400"
0 0 24 3 1 #" "
0 0 19 3 7 #"\"solid\""
0 0 24 3 1 #" "
0 0 19 3 7 #"\"green\""
0 0 24 3 2 #")]"
0 0 24 29 1 #"\n"
0 0 24 3 9 #"        ["
0 0 14 3 4 #"else"
0 0 24 3 2 #" ("
0 0 14 3 9 #"rectangle"
0 0 24 3 1 #" "
0 0 21 3 3 #"700"
0 0 24 3 1 #" "
0 0 21 3 3 #"400"
0 0 24 3 1 #" "
0 0 19 3 7 #"\"solid\""
0 0 24 3 1 #" "
0 0 19 3 5 #"\"red\""
0 0 24 3 4 #")]))"
0 0 24 29 1 #"\n"
0 0 24 29 1 #"\n"
0 0 24 3 1 #"("
0 0 15 3 6 #"define"
0 0 24 3 2 #" ("
0 0 14 3 13 #"mouse-handler"
0 0 24 3 1 #" "
0 0 14 3 5 #"model"
0 0 24 3 1 #" "
0 0 14 3 1 #"x"
0 0 24 3 1 #" "
0 0 14 3 1 #"y"
0 0 24 3 1 #" "
0 0 14 3 5 #"event"
0 0 24 3 1 #")"
0 0 24 29 1 #"\n"
0 0 24 3 3 #"  ("
0 0 15 3 4 #"cond"
0 0 24 3 3 #" [("
0 0 14 3 11 #"in-ellipse?"
0 0 24 3 1 #" "
0 0 21 3 3 #"700"
0 0 24 3 1 #" "
0 0 21 3 3 #"200"
0 0 24 3 1 #" "
0 0 21 3 3 #"350"
0 0 24 3 1 #" "
0 0 21 3 1 #"0"
0 0 24 3 1 #" "
0 0 21 3 1 #"0"
0 0 24 3 1 #" "
0 0 21 3 3 #"200"
0 0 24 3 1 #" "
0 0 14 3 1 #"x"
0 0 24 3 1 #" "
0 0 14 3 1 #"y"
0 0 24 3 1 #")"
0 0 24 29 1 #"\n"
0 0 24 3 9 #"         "
0 0 21 3 2 #"#t"
0 0 24 3 1 #"]"
0 0 24 29 1 #"\n"
0 0 24 3 9 #"        ["
0 0 14 3 4 #"else"
0 0 24 3 1 #" "
0 0 21 3 2 #"#f"
0 0 24 3 3 #"]))"
0 0 24 29 1 #"\n"
0 0 24 29 1 #"\n"
0 0 24 3 1 #"("
0 0 14 3 8 #"big-bang"
0 0 24 3 1 #" "
0 0 21 3 2 #"#t"
0 0 24 29 1 #"\n"
0 0 24 3 11 #"          ("
0 0 14 3 7 #"to-draw"
0 0 24 3 1 #" "
0 0 14 3 6 #"render"
0 0 24 3 1 #")"
0 0 24 29 1 #"\n"
0 0 24 3 11 #"          ("
0 0 14 3 8 #"on-mouse"
0 0 24 3 1 #" "
0 0 14 3 13 #"mouse-handler"
0 0 24 3 3 #")) "
0           0
